testing/enclient/enclient.go

```diff
@@ -0,0 +1,149 @@
+package enclient
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/big"
+	"net/http"
+	"time"
+
+	"github.com/google/exposure-notifications-server/internal/base64util"
+	"github.com/google/exposure-notifications-server/internal/database"
+)
+
+const (
+	dkLen               = 16
+	maxTransmissionRisk = 8
+	maxIntervals        = 144
+)
+
+type Interval int32
+
+// Posts requests to the specified url.
+// This methods attempts to serialize data argument as a json.
+func PostRequest(url string, data interface{}) (*http.Response, error) {
+	request := bytes.NewBuffer(JsonRequest(data))
+	r, err := http.NewRequest("POST", url, request)
+	if err != nil {
+		return nil, err
+	}
+	r.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
+	resp, err := client.Do(r)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return resp, fmt.Errorf("post request failed with status: %v", resp.StatusCode)
```

Comment:
You might want to copy the body here and put it in the error message. See devicecheck.go for an example.